src/DIRAC/Resources/Computing/HTCondorCEComputingElement.py

```diff
@@ -237,6 +252,38 @@ def _reset(self):
         self.log.debug("Remote scheduler option:", self.remoteScheddOptions)
         return S_OK()
 
+    def _executeCondorCommand(self, cmd, keepTokenFile=False):
+        tFile = None
+        if self.token:
+            # Create a new token file if we do not keep it across several calls
+            if keepTokenFile:
+                tFile = self.tokenFile
+            if not tFile:
+                tFile = tempfile.NamedTemporaryFile(suffix=".token", prefix="HTCondorCE_", dir=self.workingDirectory)
+                writeToTokenFile(self.token["access_token"], tFile.name)
+
+            htcEnv = {
+                "_CONDOR_SEC_CLIENT_AUTHENTICATION_METHODS": "SCITOKENS",
+                "_CONDOR_SCITOKENS_FILE": tFile.name,
+            }
+        else:
+            htcEnv = {"_CONDOR_SEC_CLIENT_AUTHENTICATION_METHODS": "GSI"}
+
+        result = executeGridCommand(
+            self.proxy,
+            cmd,
+            gridEnvScript=self.gridEnv,
+            gridEnvDict=htcEnv,
+        )
+        # Remove token file if we do not want to keep it
+        if tFile:
+            if keepTokenFile:
+                self.tokenFile = tFile
+            else:
+                self.tokenFile = None
```

Comment:
I think this can be simplified by dropping the `tFile` variable:

```suggestion
        if self.token:
            # Create a new token file if we do not keep it across several calls
            if not self.tokenFile or not keepTokenFile:
                self.tokenFile = tempfile.NamedTemporaryFile(suffix=".token", prefix="HTCondorCE_", dir=self.workingDirectory)
                writeToTokenFile(self.token["access_token"], self.tokenFile.name)

            htcEnv = {
                "_CONDOR_SEC_CLIENT_AUTHENTICATION_METHODS": "SCITOKENS",
                "_CONDOR_SCITOKENS_FILE": self.tokenFile.name,
            }
        else:
            htcEnv = {"_CONDOR_SEC_CLIENT_AUTHENTICATION_METHODS": "GSI"}

        result = executeGridCommand(
            self.proxy,
            cmd,
            gridEnvScript=self.gridEnv,
            gridEnvDict=htcEnv,
        )
        self.tokenFile = tFile if keepTokenFile else None
```

Reply:
Yes, just the last line should be rather:

     self.tokenFile = self.tokenFile if keepTokenFile else None

Although I think it is more explicit to keep tFile variable. This one is local and used in the local logic whereas self.tokenFile keeps this reference across calls. So, two different purposes. I think it is more readable. More concise is not always better for readability

Reply:
I'm making the suggestion not for making concise but for reducing the complexity of the function (as in [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity)).

Reply:
Wow, I have learnt something today, thank you. I appreciate this piece from the above reference:

"...Reducing the cyclomatic complexity of code is [not proven](https://en.wikipedia.org/wiki/Correlation_does_not_imply_causation) to reduce the number of errors or bugs in that code. International safety standards like [ISO 26262](https://en.wikipedia.org/wiki/ISO_26262), however, mandate coding guidelines that enforce low code complexity.[[16]](https://en.wikipedia.org/wiki/Cyclomatic_complexity#cite_note-ISO26262Part3-18)..."